test/TensorFlowNET.UnitTest/Training/GradientDescentOptimizerTests.cs

```diff
@@ -67,6 +68,56 @@ public void TestBasic()
             TestBasic<double>();
         }
 
+        private void TestMinimizeResourceVariable<T>() where T : struct
+        {
+            var dtype = GetTypeForNumericType<T>();
+
+            // train.GradientDescentOptimizer is V1 only API.
+            tf.Graph().as_default();
+            using (var sess = self.cached_session())
+            {
+                var var0 = tf.Variable(new[] { new[] { 1.0, 2.0 } }, dtype: dtype);
+                var var1 = tf.Variable(new[] { 3.0 }, dtype: dtype);
+                var x = tf.constant(
```

Comment:
@Oceania2018 I can't initialize such a 2D constant. It either throws conversion exception or data allocation exception. Do I do anything wrong or there is a bug?

Reply:
Is it okay to float type? I'm not sure whether tf.net can initialize a 2D constant with double type, but it absolutly can when float type.

Reply:
I just tried float type.
```
public static TensorProto make_tensor_proto(object values, TF_DataType dtype = TF_DataType.DtInvalid, Shape? shape = null, bool verify_shape = false, bool allow_broadcast = false)
{
   ...
   var len = dtype.get_datatype_size() * (int)shape.size;
   byte[] bytes = new byte[len];
   System.Buffer.BlockCopy(array, 0, bytes, 0, len);
```
`BlockCopy` expects that array contains only primitive values and throws there. 
Could you please give me an example where a float 2D constant works?

Reply:
Here is a example use float type:
https://github.com/SciSharp/TensorFlow.NET/blob/8cab730f98112a845461bb635f137e220f272adb/test/TensorFlowNET.Graph.UnitTest/ImageTest.cs#L54-L82

Reply:
I see. It should be a 2D array `[,]` , not a jagged array `[][]`. Thanks.